controllers/imageclusterinstall_controller_test.go

```diff
@@ -922,6 +922,65 @@ var _ = Describe("Reconcile", func() {
 
 	})
 
+	It("Set status condition in case dataImage exists and being deleted", func() {
+		bmh := bmhInState(bmh_v1alpha1.StateAvailable)
+		bmh.Spec.ExternallyProvisioned = true
+		Expect(c.Create(ctx, bmh)).To(Succeed())
+
+		clusterInstall.Spec.BareMetalHostRef = &v1alpha1.BareMetalHostReference{
+			Name:      bmh.Name,
+			Namespace: bmh.Namespace,
+		}
+		Expect(c.Create(ctx, clusterInstall)).To(Succeed())
+		Expect(c.Create(ctx, clusterDeployment)).To(Succeed())
+
+		dataImage := bmh_v1alpha1.DataImage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       bmh.Name,
+				Namespace:  bmh.Namespace,
+				Finalizers: []string{"test.finalizer"}, // Prevent immediate deletion
+			},
+		}
+		Expect(c.Create(ctx, &dataImage)).To(Succeed())
+
+		defer func() {
+			dataImage.SetFinalizers([]string{}) // Remove finalizer
+			Expect(c.Update(ctx, &dataImage)).To(Succeed())
+		}()
```

Comment:
Should we do this in an `After` block of some sort?
Separately is this really part of the test or some kind of cleanup?
If it's cleanup I feel like it's just not necessary since we're re-initializing the fake client for each test.